terminal.go

```diff
@@ -850,22 +855,33 @@ func (g *LightningTerminal) startInternalSubServers(
 		return nil
 	}
 
+	stopAccountService := func() {
+		if err := g.accountService.Stop(); err != nil {
+			// We only log the error if we fail to stop the service,
+			// as it's not critical that this succeeds in order to
+			// keep litd running
+			log.Errorf("Error stopping account service: %v", err)
+		}
+	}
+
 	log.Infof("Starting LiT account service")
-	err = g.accountService.Start(
-		g.lndClient.Client, g.lndClient.Router,
-		g.lndClient.ChainParams,
-	)
-	if err != nil {
-		log.Errorf("error starting account service: %v, disabling "+
-			"account service", err)
-		g.statusMgr.SetErrored(subservers.ACCOUNTS, err.Error())
+	if g.cfg.AccountsMode != ModeDisable {
+		err = g.accountService.Start(
+			g.lndClient.Client, g.lndClient.Router,
+			g.lndClient.ChainParams,
+		)
+		if err != nil {
+			log.Errorf("error starting account service: %v, "+
+				"disabling account service", err)
+			g.statusMgr.SetErrored(subservers.ACCOUNTS, err.Error())
+			stopAccountService()
+		} else {
+			g.statusMgr.SetRunning(subservers.ACCOUNTS)
+			g.accountServiceStarted = true
+		}
 	} else {
-		g.statusMgr.SetRunning(subservers.ACCOUNTS)
+		stopAccountService()
```

Comment:
This `else` seems to be wrong... Why would we stop the service if we haven't started it?

Reply:
So agree that this is a bit confusing. The reason why we need to do this is that `s.accountService.Stop()` closes the contexts and db store which are opened when we create the account service with `accounts.NewService` in the beginning of the `LightningTerminal.start` function. This is done regardless of if the accounts service is `disabled` or if we fail to start it, so we need to close the contexts and db store either way, so this is why we run the `stopAccountService` function either way.

I changed the name of the `stopAccountService` function to `closeAccountService` to better represent what it actually does to make it more understandable. Let me know though if you have a suggestion of how I could do this better :)